src/audio/module_adapter/CMakeLists.txt

```diff
@@ -5,6 +5,10 @@ if(NOT CONFIG_LIBRARY)
 	add_local_sources(sof module_adapter.c module/generic.c)
 	endif()
 
+	if(CONFIG_IPC_MAJOR_4 AND CONFIG_COMP_SEL)
+	add_local_sources(sof module_adapter.c module/generic.c)
```

Comment:
So when CONFIG_IPC_MAJOR_4=y and CONFIG_COMP_SEL=y then we
add module_adapter.c and module/generic.c to sof sources two times?
This does not seem right...
Looks like you need to update statement above to:
`if ((CONFIG_IPC_MAJOR_4 AND CONFIG_COMP_SEL) OR CONFIG_CADENCE_CODEC OR (NOT CONFIG_COMP_LEGACY_INTERFACE))`
if I understand your intentions correctly.

Reply:
thanks.

Reply:
I checked it again. I don't need to change CMakeFile for selector in ipc4 since these module_xx.c will be added in ipc4 no matter selector is enabled.